### PR TITLE
Abstract the code of a module as a kmodule().

### PR DESCRIPTION
**Pull request by aggelgian**
Starting off with an implementation of
`  kmodule() :: ets:tid()`
for compatibility with the current code, but will change to
`  kmodule() :: #{ ... }`
in a follow-up PR.

### CODE REVIEW

File: test/utest/src/cuter_codeserver_tests.erl [hunk 1]
```diff
@@ -5,17 +5,11 @@
...
 -define(MODS_LIST, [lists, dict, orddict, ets, os, string, filelib, beam_lib, cerl]).
 
--type descr() :: nonempty_string().
--type f_one() :: fun((term()) -> term()).
--type setup() :: {'setup', fun(() -> term()), f_one(), f_one()}.
--type ret_t() :: {descr(), setup()}.
-
-%% Load modules
--spec load_modules_test_() -> ret_t().
+-spec load_modules_test_() -> any().
```
Comment (kostis) [inline test/utest/src/cuter_codeserver_tests.erl:12]: This function returns a 2-tuple, with a non-empty string in its first element, not `any()`...

File: test/utest/src/cuter_codeserver_tests.erl [hunk 2]
```diff
@@ -49,9 +43,9 @@ load_and_retrieve_test_() ->
   end,
   {"Query modules multiple times", {setup, Setup, Cleanup, Inst}}.
 
-%% Get the specs of mfas.
--spec get_spec_test_() -> ret_t().
-get_spec_test_() ->
+
+-spec get_mfa_spec_test_() -> any().
```
Comment (kostis) [inline test/utest/src/cuter_codeserver_tests.erl:47]: Similar comment for this function.

File: test/utest/src/cuter_cerl_tests.erl [hunk 3]
```diff
@@ -107,3 +72,20 @@ construct_and_access_kfun_test() ->
   GotCode = cuter_cerl:kfun_code(Kfun),
   GotIsExported = cuter_cerl:kfun_is_exported(Kfun),
   [?assertEqual(Code, GotCode), ?assertEqual(true, GotIsExported)].
+
+%% -------------------------------------------------------------------
+%% kmodule API
+%% -------------------------------------------------------------------
+
+-spec construct_and_access_kmodule_test() -> ok.
```
Comment (kostis) [inline test/utest/src/cuter_cerl_tests.erl:80]: This function does not return `ok`,

File: test/utest/src/cuter_cerl_tests.erl [hunk 4]
```diff
@@ -6,60 +6,43 @@
...
--spec load_exports_test_() -> [ret_t()].
-load_exports_test_() ->
-  Setup = fun(M) -> fun() -> setup(M) end end,
-  Cleanup = fun cleanup/1,
-  Inst = fun load_exports/1,
-  [{"Validate exported funs: " ++ atom_to_list(M),
-    {setup, Setup(M), Cleanup, Inst}} || M <- ?MODS_LIST].
-  
-load_exports({M, MDb}) ->
+-spec exported_mfas_in_lists_module_test_() -> any().
```
Comment (kostis) [inline test/utest/src/cuter_cerl_tests.erl:18]: This function does not return `any()`; it returns a list of terms (of some type).

File: test/utest/src/cuter_cerl_tests.erl [hunk 5]
```diff
@@ -6,60 +6,43 @@
 -include("include/eunit_config.hrl").
 -include("include/cuter_macros.hrl").
 
--spec test() -> 'ok' | {'error', term()}.  %% This should be provided by EUnit
+-spec test() -> any().
```
Comment (kostis) [inline test/utest/src/cuter_cerl_tests.erl:9]: Why lie (overapproximate) in the return type?

**Review by kostis — COMMENTED** [review]
Looks OK overall, but I do not like the over-approximations in the types.

File: test/utest/src/cuter_codeserver_tests.erl (hunk 1 above)
Comment (aggelgian) [reply test/utest/src/cuter_codeserver_tests.erl]: Added an include file and moved these typedefs there. Now we can reuse them for all utest modules.

File: test/utest/src/cuter_codeserver_tests.erl (hunk 2 above)
Comment (aggelgian) [reply test/utest/src/cuter_codeserver_tests.erl]: Done.

File: test/utest/src/cuter_cerl_tests.erl (hunk 3 above)
Comment (aggelgian) [reply test/utest/src/cuter_cerl_tests.erl]: Done.

File: test/utest/src/cuter_cerl_tests.erl (hunk 5 above)
Comment (aggelgian) [reply test/utest/src/cuter_cerl_tests.erl]: Done.

**Comment by aggelgian** [comment]
> Looks OK overall, but I do not like the over-approximations in the types.

Fixed these issues. Right now we are not running dialyzer on utests.
We may want to do so in the future to put those specs in use.

**Pull request merged**